src/Core/FlatpakBackend.vala

```diff
@@ -405,7 +405,38 @@ public class AppCenterCore.FlatpakBackend : Backend, Object {
     }
 
     public Gee.Collection<Package> get_packages_by_author (string author, int max) {
-        return new Gee.ArrayList<Package> ();
+        var packages = new Gee.ArrayList<AppCenterCore.Package> ();
+        var package_ids = new Gee.ArrayList<string> ();
+
+        foreach (var package in package_list.values) {
+            if (packages.size > max) {
+                break;
+            }
+
+            if (package.component.id in package_ids) {
+                break;
```

Comment:
This feels like it shouldn't be a `break` but a `continue`? Or else you'll end up bailing out if the same package id comes up twice, rather than when you've found as many as you need?

Reply:
Ah yes, thanks for catching that!!